### DISCUSSION TITLE
Hide gauges for queries that return no results

### DISCUSSION THREAD

**Discussion by brunofrohlich** [grafana/grafana#41300] [Dashboards, Panels & Visualizations]
**What happened**:
Hello, I updated my Grafana for the last version and now I'm experiencing an issue that wasn't present in previous versions.

I have a panel with Gauge visualization with 4 queries and each one has a different target, as you can see in the screenshot below.

![image](https://user-images.githubusercontent.com/74659068/128398066-76f9e716-840b-4d71-ab86-1824ac557b12.png)

**What you expected to happen**:
In the previous versions of Grafana, the gauges with no query results used to be hidden and only the one with results would appear. Now it seems that this configuration was disabled and all gauges are being shown even if null.

**How to reproduce it (as minimally and precisely as possible)**:
Create 2 different queries with different targets and just one with results.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.0.6
- Data source type & version: PostgreSQL 10
- OS Grafana is installed on: Windows
- User OS & Browser: Edge
- Grafana plugins:
- Others:

**Comment by torkelo** [comment]
Can you show your queries and the results? (Panel would help and query inspector results)

**Comment by brunofrohlich** [comment]
A:1 rows
SELECT
(SELECT SUM(scrap) 
FROM
(
SELECT 
CASE
WHEN p.mat_production_group in ('AD01') then 'TRUCK X-PLY'
WHEN p.mat_production_group in ('AC11') then 'LIGHT TRUCK'
WHEN p.mat_production_group in ('FB00') then 'ALL STEEL'
WHEN p.mat_production_group in ('AG','AI','BA4','BB00','AH11','AG11','AF11') then 'AGRO'
end
BU,
SUM (fdd_qty) AS scrap
  FROM BR_GV_CAR_FAC_PCS02.fin_defects_declared as f
      INNER JOIN BR_GV_CAR_FAC_PCS02.md_machines AS m
      ON m.mch_code = f.fdd_mch_curing_code
      INNER JOIN BR_GV_CAR_FAC_PCS02.md_materials AS p
      ON f.mat_sap_code = p.mat_sap_code AND f.cnt_code = p.cnt_code AND f.mat_variant = p.mat_variant
          WHERE fdd_active = 'T'
          AND fds_severity in ('10','20')
          AND m.mch_dc_server in ('J08','J03')
          AND fdd_date BETWEEN '2021-07-31T23:59:59Z' AND '2021-08-04T23:59:58.999Z'
          AND FPT_CODE <>'PT'
          AND f.mat_sap_code like 'SAW%'
          GROUP BY p.mat_production_group
)B
WHERE B.BU = 'ALL STEEL' AND B.BU IN ('ALL STEEL')

 
     )
/
(SELECT SUM(production)
FROM
(
SELECT 
CASE
WHEN p.mat_production_group in ('AD01') then 'TRUCK X-PLY'
WHEN p.mat_production_group in ('AC11') then 'LIGHT TRUCK'
WHEN p.mat_production_group in ('FB00') then 'ALL STEEL'
WHEN p.mat_production_group in ('AG','AI','BA4','BB00','AH11','AG11','AF11') then 'AGRO'
end
BU,
SUM (ps_quantity) AS production
  FROM BR_GV_CAR_FAC_PCS02.dc_production_data as v
      INNER JOIN BR_GV_CAR_FAC_PCS02.md_machines AS m
      ON m.mch_code = v.mch_code
      INNER JOIN BR_GV_CAR_FAC_PCS02.md_materials AS p
      ON v.mat_sap_code = p.mat_sap_code AND v.cnt_code = p.cnt_code AND v.mat_variant = p.mat_variant
          WHERE v.pp_code = 'vul'
          AND m.mch_dc_server in ('J08','J03')
          AND ps_date BETWEEN '2021-07-31T23:59:59Z' AND '2021-08-04T23:59:58.999Z'
          GROUP BY p.mat_production_group
)B
WHERE B.BU = 'ALL STEEL' AND B.BU IN ('ALL STEEL'))*100 AS SCRAP_AS
B:1 rows
SELECT
(SELECT SUM(scrap) 
FROM
(
SELECT 
CASE
WHEN p.mat_production_group in ('AD01') then 'TRUCK X-PLY'
WHEN p.mat_production_group in ('AC11') then 'LIGHT TRUCK'
WHEN p.mat_production_group in ('FB00') then 'ALL STEEL'
WHEN p.mat_production_group in ('AG','AI','BA4','BB00','AH11','AG11','AF11') then 'AGRO'
end
BU,
SUM (fdd_qty) AS scrap
  FROM BR_GV_CAR_FAC_PCS02.fin_defects_declared as f
      INNER JOIN BR_GV_CAR_FAC_PCS02.md_machines AS m
      ON m.mch_code = f.fdd_mch_curing_code
      INNER JOIN BR_GV_CAR_FAC_PCS02.md_materials AS p
      ON f.mat_sap_code = p.mat_sap_code AND f.cnt_code = p.cnt_code AND f.mat_variant = p.mat_variant
          WHERE fdd_active = 'T'
          AND fds_severity in ('10','20')
          AND m.mch_dc_server in ('J08','J03')
          AND fdd_date BETWEEN '2021-07-31T23:59:59Z' AND '2021-08-04T23:59:58.999Z'
          AND FPT_CODE <>'PT'
          GROUP BY p.mat_production_group
)B
WHERE B.BU = 'AGRO' AND B.BU IN ('ALL STEEL')

 
     )
/
(SELECT SUM(production)
FROM
(
SELECT 
CASE
WHEN p.mat_production_group in ('AD01') then 'TRUCK X-PLY'
WHEN p.mat_production_group in ('AC11') then 'LIGHT TRUCK'
WHEN p.mat_production_group in ('FB00') then 'ALL STEEL'
WHEN p.mat_production_group in ('AG','AI','BA4','BB00','AH11','AG11','AF11') then 'AGRO'
end
BU,
SUM (ps_quantity) AS production
  FROM BR_GV_CAR_FAC_PCS02.dc_production_data as v
      INNER JOIN BR_GV_CAR_FAC_PCS02.md_machines AS m
      ON m.mch_code = v.mch_code
      INNER JOIN BR_GV_CAR_FAC_PCS02.md_materials AS p
      ON v.mat_sap_code = p.mat_sap_code AND v.cnt_code = p.cnt_code AND v.mat_variant = p.mat_variant
          WHERE v.pp_code = 'vul'
          AND m.mch_dc_server in ('J08','J03')
          AND ps_date BETWEEN '2021-07-31T23:59:59Z' AND '2021-08-04T23:59:58.999Z'
          GROUP BY p.mat_production_group
)B
WHERE B.BU = 'AGRO' AND B.BU IN ('ALL STEEL'))*100 AS SCRAP_AG

 
     
C:1 rows
SELECT
(SELECT SUM(scrap) 
FROM
(
SELECT 
CASE
WHEN p.mat_production_group in ('AD01') then 'TRUCK X-PLY'
WHEN p.mat_production_group in ('AC11') then 'LIGHT TRUCK'
WHEN p.mat_production_group in ('FB00') then 'ALL STEEL'
WHEN p.mat_production_group in ('AG','AI','BA4','BB00','AH11','AG11','AF11') then 'AGRO'
end
BU,
SUM (fdd_qty) AS scrap
  FROM BR_GV_CAR_FAC_PCS02.fin_defects_declared as f
      INNER JOIN BR_GV_CAR_FAC_PCS02.md_machines AS m
      ON m.mch_code = f.fdd_mch_curing_code
      INNER JOIN BR_GV_CAR_FAC_PCS02.md_materials AS p
      ON f.mat_sap_code = p.mat_sap_code AND f.cnt_code = p.cnt_code AND f.mat_variant = p.mat_variant
          WHERE fdd_active = 'T'
          AND fds_severity in ('10','20')
          AND m.mch_dc_server in ('J08','J03')
          AND fdd_date BETWEEN '2021-07-31T23:59:59Z' AND '2021-08-04T23:59:58.999Z'
          AND FPT_CODE <>'PT'
          GROUP BY p.mat_production_group
)B
WHERE B.BU = 'TRUCK X-PLY' AND B.BU IN ('ALL STEEL')

 
     )
/
(SELECT SUM(production)
FROM
(
SELECT 
CASE
WHEN p.mat_production_group in ('AD01') then 'TRUCK X-PLY'
WHEN p.mat_production_group in ('AC11') then 'LIGHT TRUCK'
WHEN p.mat_production_group in ('FB00') then 'ALL STEEL'
WHEN p.mat_production_group in ('AG','AI','BA4','BB00','AH11','AG11','AF11') then 'AGRO'
end
BU,
SUM (ps_quantity) AS production
  FROM BR_GV_CAR_FAC_PCS02.dc_production_data as v
      INNER JOIN BR_GV_CAR_FAC_PCS02.md_machines AS m
      ON m.mch_code = v.mch_code
      INNER JOIN BR_GV_CAR_FAC_PCS02.md_materials AS p
      ON v.mat_sap_code = p.mat_sap_code AND v.cnt_code = p.cnt_code AND v.mat_variant = p.mat_variant
          WHERE v.pp_code = 'vul'
          AND m.mch_dc_server in ('J08','J03')
          AND ps_date BETWEEN '2021-07-31T23:59:59Z' AND '2021-08-04T23:59:58.999Z'
          GROUP BY p.mat_production_group
)B
WHERE B.BU = 'TRUCK X-PLY' AND B.BU IN ('ALL STEEL'))*100 AS SCRAP_XP
D:1 rows
SELECT
(SELECT SUM(scrap) 
FROM
(
SELECT 
CASE
WHEN p.mat_production_group in ('AD01') then 'TRUCK X-PLY'
WHEN p.mat_production_group in ('AC11') then 'LIGHT TRUCK'
WHEN p.mat_production_group in ('FB00') then 'ALL STEEL'
WHEN p.mat_production_group in ('AG','AI','BA4','BB00','AH11','AG11','AF11') then 'AGRO'
end
BU,
SUM (fdd_qty) AS scrap
  FROM BR_GV_CAR_FAC_PCS02.fin_defects_declared as f
      INNER JOIN BR_GV_CAR_FAC_PCS02.md_machines AS m
      ON m.mch_code = f.fdd_mch_curing_code
      INNER JOIN BR_GV_CAR_FAC_PCS02.md_materials AS p
      ON f.mat_sap_code = p.mat_sap_code AND f.cnt_code = p.cnt_code AND f.mat_variant = p.mat_variant
          WHERE fdd_active = 'T'
          AND fds_severity in ('10','20')
          AND m.mch_dc_server in ('J08','J03')
          AND fdd_date BETWEEN '2021-07-31T23:59:59Z' AND '2021-08-04T23:59:58.999Z'
          AND FPT_CODE <>'PT'
          GROUP BY p.mat_production_group
)B
WHERE B.BU = 'LIGHT TRUCK' AND B.BU IN ('ALL STEEL')

 
     )
/
(SELECT SUM(production)
FROM
(
SELECT 
CASE
WHEN p.mat_production_group in ('AD01') then 'TRUCK X-PLY'
WHEN p.mat_production_group in ('AC11') then 'LIGHT TRUCK'
WHEN p.mat_production_group in ('FB00') then 'ALL STEEL'
WHEN p.mat_production_group in ('AG','AI','BA4','BB00','AH11','AG11','AF11') then 'AGRO'
end
BU,
SUM (ps_quantity) AS production
  FROM BR_GV_CAR_FAC_PCS02.dc_production_data as v
      INNER JOIN BR_GV_CAR_FAC_PCS02.md_machines AS m
      ON m.mch_code = v.mch_code
      INNER JOIN BR_GV_CAR_FAC_PCS02.md_materials AS p
      ON v.mat_sap_code = p.mat_sap_code AND v.cnt_code = p.cnt_code AND v.mat_variant = p.mat_variant
          WHERE v.pp_code = 'vul'
          AND m.mch_dc_server in ('J08','J03')
          AND ps_date BETWEEN '2021-07-31T23:59:59Z' AND '2021-08-04T23:59:58.999Z'
          GROUP BY p.mat_production_group
)B
WHERE B.BU = 'LIGHT TRUCK' AND B.BU IN ('ALL STEEL'))*100 AS SCRAP_LT

**Comment by brunofrohlich** [comment]
> Can you show your queries and the results? (Panel would help and query inspector results)
Okay.
It's basically one query for each filter value, because each one has a fixed target and I cannot put in one gauge only.

The results are shown in the screenshot above.

**Comment by torkelo** [comment]
looks like the results are 0 which is not the same as no result

**Comment by brunofrohlich** [comment]
Actually, is not 0, because I put 'No Value' as 0 in the panel options. If I remove, it looks like this. And the gauge still there.

In 7.5.3 version that I was previously using, it wouldn't happen.

![image](https://user-images.githubusercontent.com/74659068/128408160-3bf22022-a109-4a3d-bbda-258c0380ef63.png)

![image](https://user-images.githubusercontent.com/74659068/128408446-a03c8de2-0409-4059-8cdf-4357f4a34c94.png)

**Comment by baur** [comment]
Hi @brunofrohlich 

I'm using templating to hide empty panels, but I cant control panel width in this case

@torkelo, it would be nice add otion to save panel width as maxWidth = screenWidth / (Max per row)

![image](https://user-images.githubusercontent.com/2898361/128467962-94e989b0-b7a1-4b28-80f0-1a534415d279.png)

**Comment by torkelo** [comment]
But what you showed was not a repeated panel but a single panel with many queries? Or a single query with many sub queries? Can you show the sql results of your queries?

**Comment by brunofrohlich** [comment]
> But what you showed was not a repeated panel but a single panel with many queries? Or a single query with many sub queries? Can you show the sql results of your queries?

Yes @torkelo, it's not a repeated panel. It's a single panel with 4 queries. I'm doing like this because each query result has a different target threshold and this was the only way I could do it without putting the same threshold for every query.

Basically, with filter selection, only one query will have results at time, the other 3 ones will be equal to _null_. In Grafana 7 it was working fine, not showing the gauges with null results, but when I upgraded to Grafana 8 seems that it has changed.

Here's a snapshot of Grafana 7.5.10 with the same queries and configurations.

**Grafana 7.5.10**
![image](https://user-images.githubusercontent.com/74659068/128513716-689b730f-4ae0-4bb8-b19b-7d81a33e0afb.png)

@torkelo I think you can reproduce it in Grafana 8. Create 2 queries and make just one with results, the other one must result _null_ in Gauge visualization.

Thanks for the quick replies and support so far :)

**Comment by zuchka** [comment]
@brunofrohlich strangely, this user reported the exact opposite behavior on 8.0.6.

https://community.grafana.com/t/gauge-panel-do-not-display-gauge-if-null-empty-value/50793

On the off chance that this got fixed already, have you tried updating to 8.1.5?

**Comment by brunofrohlich** [comment]
@zuchka, that's quite strange because it's exactly what I want.

I tried 8.1.5 version now, but got the same behavior. Query A with results, and B,C,D with _NULL_ value being shown.

Anyway, I managed to solve it with another visualization, but the issue is still on.

![image](https://user-images.githubusercontent.com/74659068/134506384-d1303859-8403-4c66-96b6-ebee7c10f7fc.png)

**Comment by zuchka** [comment]
Thanks @brunofrohlich

One last thing: can you please append your panel JSON inside a code block? and maybe the raw request and response, also from the `inspect` drawer? Anything to help the squads investigate 👍 

thanks again for working with us on this

**Comment by brunofrohlich** [comment]
@zuchka For sure, here we go. Thanks for the help!
```
{
  "id": 183,
  "gridPos": {
    "h": 6,
    "w": 3,
    "x": 13,
    "y": 13
  },
  "type": "gauge",
  "title": "SCRAP MONTH TO DATE",
  "transformations": [],
  "datasource": "ZEUS_GV",
  "pluginVersion": "8.1.5",
  "timeShift": "1s",
  "hideTimeOverride": true,
  "fieldConfig": {
    "defaults": {
      "thresholds": {
        "mode": "absolute",
        "steps": [
          {
            "color": "dark-green",
            "value": null
          },
          {
            "color": "dark-yellow",
            "value": 0.92
          },
          {
            "color": "dark-red",
            "value": 1
          }
        ]
      },
      "mappings": [],
      "decimals": 2,
      "max": 2,
      "min": 0,
      "unit": "percent"
    },
    "overrides": [
      {
        "matcher": {
          "id": "byName",
          "options": "SCRAP_AS"
        },
        "properties": [
          {
            "id": "thresholds",
            "value": {
              "mode": "absolute",
              "steps": [
                {
                  "color": "dark-green",
                  "value": null
                },
                {
                  "color": "dark-red",
                  "value": 0.95
                }
              ]
            }
          },
          {
            "id": "max",
            "value": 1.8
          }
        ]
      },
      {
        "matcher": {
          "id": "byName",
          "options": "SCRAP_AG"
        },
        "properties": [
          {
            "id": "thresholds",
            "value": {
              "mode": "absolute",
              "steps": [
                {
                  "color": "green",
                  "value": null
                },
                {
                  "color": "dark-red",
                  "value": 1.04
                }
              ]
            }
          },
          {
            "id": "max",
            "value": 1.5
          }
        ]
      },
      {
        "matcher": {
          "id": "byName",
          "options": "SCRAP_XP"
        },
        "properties": [
          {
            "id": "thresholds",
            "value": {
              "mode": "absolute",
              "steps": [
                {
                  "color": "dark-green",
                  "value": null
                },
                {
                  "color": "dark-red",
                  "value": 0.39
                }
              ]
            }
          },
          {
            "id": "max",
            "value": 0.6
          }
        ]
      },
      {
        "matcher": {
          "id": "byName",
          "options": "SCRAP_LT"
        },
        "properties": [
          {
            "id": "thresholds",
            "value": {
              "mode": "absolute",
              "steps": [
                {
                  "color": "dark-green",
                  "value": null
                },
                {
                  "color": "dark-red",
                  "value": 0.39
                }
              ]
            }
          },
          {
            "id": "max",
            "value": 0.8
          }
        ]
      }
    ]
  },
  "options": {
    "reduceOptions": {
      "values": false,
      "calcs": [
        "sum"
      ],
      "fields": ""
    },
    "showThresholdLabels": true,
    "showThresholdMarkers": true,
    "text": {},
    "orientation": "auto"
  },
  "targets": [
    {
      "format": "table",
      "group": [],
      "hide": false,
      "metricColumn": "none",
      "rawQuery": true,
      "rawSql": "SELECT\n(SELECT SUM(scrap) \nFROM\n(\nSELECT \nCASE\nWHEN p.mat_production_group in ('AD01') then 'TRUCK X-PLY'\nWHEN p.mat_production_group in ('AC11') then 'LIGHT TRUCK'\nWHEN p.mat_production_group in ('FB00') then 'ALL STEEL'\nWHEN p.mat_production_group in ('AG','AI','BA4','BB00','AH11','AG11','AF11') then 'AGRO'\nend\nBU,\nSUM (fdd_qty) AS scrap\n  FROM BR_GV_CAR_FAC_PCS02.fin_defects_declared as f\n      INNER JOIN BR_GV_CAR_FAC_PCS02.md_machines AS m\n      ON m.mch_code = f.fdd_mch_curing_code\n      INNER JOIN BR_GV_CAR_FAC_PCS02.md_materials AS p\n      ON f.mat_sap_code = p.mat_sap_code AND f.cnt_code = p.cnt_code AND f.mat_variant = p.mat_variant\n          WHERE fdd_active = 'T'\n          AND fds_severity in ('10','20')\n          AND m.mch_dc_server in ('J08','J03')\n          AND $__timeFilter(fdd_date)\n          AND FPT_CODE <>'PT'\n          AND f.mat_sap_code like 'SAW%'\n          GROUP BY p.mat_production_group\n)B\nWHERE B.BU = 'ALL STEEL' AND B.BU IN ('${BU2}')\n\n \n     )\n/\n(SELECT SUM(production)\nFROM\n(\nSELECT \nCASE\nWHEN p.mat_production_group in ('AD01') then 'TRUCK X-PLY'\nWHEN p.mat_production_group in ('AC11') then 'LIGHT TRUCK'\nWHEN p.mat_production_group in ('FB00') then 'ALL STEEL'\nWHEN p.mat_production_group in ('AG','AI','BA4','BB00','AH11','AG11','AF11') then 'AGRO'\nend\nBU,\nSUM (ps_quantity) AS production\n  FROM BR_GV_CAR_FAC_PCS02.dc_production_data as v\n      INNER JOIN BR_GV_CAR_FAC_PCS02.md_machines AS m\n      ON m.mch_code = v.mch_code\n      INNER JOIN BR_GV_CAR_FAC_PCS02.md_materials AS p\n      ON v.mat_sap_code = p.mat_sap_code AND v.cnt_code = p.cnt_code AND v.mat_variant = p.mat_variant\n          WHERE v.pp_code = 'vul'\n          AND m.mch_dc_server in ('J08','J03')\n          AND $__timeFilter(ps_date)\n          GROUP BY p.mat_production_group\n)B\nWHERE B.BU = 'ALL STEEL' AND B.BU IN ('${BU2}'))*100 AS SCRAP_AS\n",
      "refId": "A",
      "select": [
        [
          {
            "params": [
              "value"
            ],
            "type": "column"
          }
        ]
      ],
      "timeColumn": "time",
      "where": [
        {
          "name": "$__timeFilter",
          "params": [],
          "type": "macro"
        }
      ]
    },
    {
      "format": "table",
      "group": [],
      "hide": false,
      "metricColumn": "none",
      "rawQuery": true,
      "rawSql": "SELECT\n(SELECT SUM(scrap) \nFROM\n(\nSELECT \nCASE\nWHEN p.mat_production_group in ('AD01') then 'TRUCK X-PLY'\nWHEN p.mat_production_group in ('AC11') then 'LIGHT TRUCK'\nWHEN p.mat_production_group in ('FB00') then 'ALL STEEL'\nWHEN p.mat_production_group in ('AG','AI','BA4','BB00','AH11','AG11','AF11') then 'AGRO'\nend\nBU,\nSUM (fdd_qty) AS scrap\n  FROM BR_GV_CAR_FAC_PCS02.fin_defects_declared as f\n      INNER JOIN BR_GV_CAR_FAC_PCS02.md_machines AS m\n      ON m.mch_code = f.fdd_mch_curing_code\n      INNER JOIN BR_GV_CAR_FAC_PCS02.md_materials AS p\n      ON f.mat_sap_code = p.mat_sap_code AND f.cnt_code = p.cnt_code AND f.mat_variant = p.mat_variant\n          WHERE fdd_active = 'T'\n          AND fds_severity in ('10','20')\n          AND m.mch_dc_server in ('J08','J03')\n          AND $__timeFilter(fdd_date)\n          AND FPT_CODE <>'PT'\n          GROUP BY p.mat_production_group\n)B\nWHERE B.BU = 'AGRO' AND B.BU IN ('${BU2}')\n\n \n     )\n/\n(SELECT SUM(production)\nFROM\n(\nSELECT \nCASE\nWHEN p.mat_production_group in ('AD01') then 'TRUCK X-PLY'\nWHEN p.mat_production_group in ('AC11') then 'LIGHT TRUCK'\nWHEN p.mat_production_group in ('FB00') then 'ALL STEEL'\nWHEN p.mat_production_group in ('AG','AI','BA4','BB00','AH11','AG11','AF11') then 'AGRO'\nend\nBU,\nSUM (ps_quantity) AS production\n  FROM BR_GV_CAR_FAC_PCS02.dc_production_data as v\n      INNER JOIN BR_GV_CAR_FAC_PCS02.md_machines AS m\n      ON m.mch_code = v.mch_code\n      INNER JOIN BR_GV_CAR_FAC_PCS02.md_materials AS p\n      ON v.mat_sap_code = p.mat_sap_code AND v.cnt_code = p.cnt_code AND v.mat_variant = p.mat_variant\n          WHERE v.pp_code = 'vul'\n          AND m.mch_dc_server in ('J08','J03')\n          AND $__timeFilter(ps_date)\n          GROUP BY p.mat_production_group\n)B\nWHERE B.BU = 'AGRO' AND B.BU IN ('${BU2}'))*100 AS SCRAP_AG\n\n \n     ",
      "refId": "B",
      "select": [
        [
          {
            "params": [
              "value"
            ],
            "type": "column"
          }
        ]
      ],
      "timeColumn": "time",
      "where": [
        {
          "name": "$__timeFilter",
          "params": [],
          "type": "macro"
        }
      ]
    },
    {
      "format": "table",
      "group": [],
      "hide": false,
      "metricColumn": "none",
      "rawQuery": true,
      "rawSql": "SELECT\n(SELECT SUM(scrap) \nFROM\n(\nSELECT \nCASE\nWHEN p.mat_production_group in ('AD01') then 'TRUCK X-PLY'\nWHEN p.mat_production_group in ('AC11') then 'LIGHT TRUCK'\nWHEN p.mat_production_group in ('FB00') then 'ALL STEEL'\nWHEN p.mat_production_group in ('AG','AI','BA4','BB00','AH11','AG11','AF11') then 'AGRO'\nend\nBU,\nSUM (fdd_qty) AS scrap\n  FROM BR_GV_CAR_FAC_PCS02.fin_defects_declared as f\n      INNER JOIN BR_GV_CAR_FAC_PCS02.md_machines AS m\n      ON m.mch_code = f.fdd_mch_curing_code\n      INNER JOIN BR_GV_CAR_FAC_PCS02.md_materials AS p\n      ON f.mat_sap_code = p.mat_sap_code AND f.cnt_code = p.cnt_code AND f.mat_variant = p.mat_variant\n          WHERE fdd_active = 'T'\n          AND fds_severity in ('10','20')\n          AND m.mch_dc_server in ('J08','J03')\n          AND $__timeFilter(fdd_date)\n          AND FPT_CODE <>'PT'\n          GROUP BY p.mat_production_group\n)B\nWHERE B.BU = 'TRUCK X-PLY' AND B.BU IN ('${BU2}')\n\n \n     )\n/\n(SELECT SUM(production)\nFROM\n(\nSELECT \nCASE\nWHEN p.mat_production_group in ('AD01') then 'TRUCK X-PLY'\nWHEN p.mat_production_group in ('AC11') then 'LIGHT TRUCK'\nWHEN p.mat_production_group in ('FB00') then 'ALL STEEL'\nWHEN p.mat_production_group in ('AG','AI','BA4','BB00','AH11','AG11','AF11') then 'AGRO'\nend\nBU,\nSUM (ps_quantity) AS production\n  FROM BR_GV_CAR_FAC_PCS02.dc_production_data as v\n      INNER JOIN BR_GV_CAR_FAC_PCS02.md_machines AS m\n      ON m.mch_code = v.mch_code\n      INNER JOIN BR_GV_CAR_FAC_PCS02.md_materials AS p\n      ON v.mat_sap_code = p.mat_sap_code AND v.cnt_code = p.cnt_code AND v.mat_variant = p.mat_variant\n          WHERE v.pp_code = 'vul'\n          AND m.mch_dc_server in ('J08','J03')\n          AND $__timeFilter(ps_date)\n          GROUP BY p.mat_production_group\n)B\nWHERE B.BU = 'TRUCK X-PLY' AND B.BU IN ('${BU2}'))*100 AS SCRAP_XP\n",
      "refId": "C",
      "select": [
        [
          {
            "params": [
              "value"
            ],
            "type": "column"
          }
        ]
      ],
      "timeColumn": "time",
      "where": [
        {
          "name": "$__timeFilter",
          "params": [],
          "type": "macro"
        }
      ]
    },
    {
      "format": "table",
      "group": [],
      "metricColumn": "none",
      "rawQuery": true,
      "rawSql": "SELECT\n(SELECT SUM(scrap) \nFROM\n(\nSELECT \nCASE\nWHEN p.mat_production_group in ('AD01') then 'TRUCK X-PLY'\nWHEN p.mat_production_group in ('AC11') then 'LIGHT TRUCK'\nWHEN p.mat_production_group in ('FB00') then 'ALL STEEL'\nWHEN p.mat_production_group in ('AG','AI','BA4','BB00','AH11','AG11','AF11') then 'AGRO'\nend\nBU,\nSUM (fdd_qty) AS scrap\n  FROM BR_GV_CAR_FAC_PCS02.fin_defects_declared as f\n      INNER JOIN BR_GV_CAR_FAC_PCS02.md_machines AS m\n      ON m.mch_code = f.fdd_mch_curing_code\n      INNER JOIN BR_GV_CAR_FAC_PCS02.md_materials AS p\n      ON f.mat_sap_code = p.mat_sap_code AND f.cnt_code = p.cnt_code AND f.mat_variant = p.mat_variant\n          WHERE fdd_active = 'T'\n          AND fds_severity in ('10','20')\n          AND m.mch_dc_server in ('J08','J03')\n          AND $__timeFilter(fdd_date)\n          AND FPT_CODE <>'PT'\n          GROUP BY p.mat_production_group\n)B\nWHERE B.BU = 'LIGHT TRUCK' AND B.BU IN ('${BU2}')\n\n \n     )\n/\n(SELECT SUM(production)\nFROM\n(\nSELECT \nCASE\nWHEN p.mat_production_group in ('AD01') then 'TRUCK X-PLY'\nWHEN p.mat_production_group in ('AC11') then 'LIGHT TRUCK'\nWHEN p.mat_production_group in ('FB00') then 'ALL STEEL'\nWHEN p.mat_production_group in ('AG','AI','BA4','BB00','AH11','AG11','AF11') then 'AGRO'\nend\nBU,\nSUM (ps_quantity) AS production\n  FROM BR_GV_CAR_FAC_PCS02.dc_production_data as v\n      INNER JOIN BR_GV_CAR_FAC_PCS02.md_machines AS m\n      ON m.mch_code = v.mch_code\n      INNER JOIN BR_GV_CAR_FAC_PCS02.md_materials AS p\n      ON v.mat_sap_code = p.mat_sap_code AND v.cnt_code = p.cnt_code AND v.mat_variant = p.mat_variant\n          WHERE v.pp_code = 'vul'\n          AND m.mch_dc_server in ('J08','J03')\n          AND $__timeFilter(ps_date)\n          GROUP BY p.mat_production_group\n)B\nWHERE B.BU = 'LIGHT TRUCK' AND B.BU IN ('${BU2}'))*100 AS SCRAP_LT\n",
      "refId": "D",
      "select": [
        [
          {
            "params": [
              "value"
            ],
            "type": "column"
          }
        ]
      ],
      "timeColumn": "time",
      "where": [
        {
          "name": "$__timeFilter",
          "params": [],
          "type": "macro"
        }
      ]
    }
  ],
  "timeFrom": null,
  "transparent": true
}
```

**Comment by jackw** [comment]
I've been trying to reproduce this issue and the only thing I can find is a difference in what is hidden when `null` is passed to the gauge for a certain query. In 7.5 it seems that `null` would prevent rendering any other subsequent queries (as gauges) where as in 8 it only hides the associated gauge.

| Grafana 7.5 | Grafana 8 |
| --- | --- |
| <img width="600" alt="grafana-7 5-gauge" src="https://user-images.githubusercontent.com/73201/135065004-76d312d7-17d7-40fa-901c-9a63202644b5.png"> | <img width="600" alt="grafana-8-gauge" src="https://user-images.githubusercontent.com/73201/135065019-d16d602d-040c-4b38-9d39-81372765dfcb.png"> |

Is this the issue you're having @brunofrohlich?

**Comment by brunofrohlich** [comment]
Hello @jackw!

This is exactly what I am experiencing! Maybe for a few users this is feasible, but I think it's a good way to have different targets for the same visualization without compromising all the rest.

Thanks for the support!

**Comment by natellium** [comment]
Hey there! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

I will be closing this discussion, but if you’re still interested in this request being implemented please don’t hesitate to open a [feature request](https://github.com/grafana/grafana/issues/new?assignees=&labels=&projects=&template=1-feature_requests.md). Sorry for the inconvenience, and thanks for your contributions!

cc @grafana/dataviz-squad
